src/hardware/sblaster.cpp

```diff
@@ -1843,15 +1762,27 @@ static void dsp_do_command()
 		const bool was_asleep = sblaster->MaybeWakeUp();
 		if (changed_modes || was_asleep) {
 			// If we're waking up, then the DAC hasn't been running (or maybe
-			// wasn't running at all), so start with a fresh DAC state and
-			// ensure we're using per-frame callback timing.
+			// wasn't running at all), so start with a fresh DAC state.
 			sb.dac = {};
-			callback_type.SetPerFrame();
 		}
 
 		if (const auto dac_rate_hz = sb.dac.MeasureDacRateHz(); dac_rate_hz) {
 			sblaster->SetChannelRateHz(*dac_rate_hz);
 		}
+
+		sblaster->output_queue.NonblockingEnqueue(sb.dac.RenderFrame());
+
+		// Stopping the queue here makes the mixer dequeue nonblocking.
+		// Direct DAC mode has no set sample rate and it varies quite frequently.
+		// Without this call, I was getting stutters in OPL music in Alone in the Dark.
+
+		// If there are reports of glitchy DAC sound, we may want to revisit this.
+		// The previous approaches of per-frame callbacks and audio stretching solved this by matching to the mixer's sample rate.
+		// The advantage here is that we guarantee to never block OPL or MIDI music channels.
+
+		// It's a bit of a trade-off but with my sample size of 1 game, this sounds fine.
+		// Direct DAC on the SoundBlaster is very rarely used.
```

Comment:
Nice one, and direct DAC games are pretty rare anyway. Maybe some obscure SB 1.0 era games use it that not many people care about, dunno... Great comment 👍🏻